import_events.py

```diff
@@ -0,0 +1,236 @@
+)
+
+# The temporary table receives raw data from S3.
+# The permenant table then receives data appropriately typed.
+TABLE_NAMES = {
+    "temp":"temporary_raw_{event_type}_data",
+    "perm":"{event_type}_events{suffix}"
+}
+
+Q_DROP_TEMPORARY_TABLE = """
+    DROP TABLE IF EXISTS {table};
+""".format(table=TABLE_NAMES["temp"])
+
+Q_CREATE_EVENTS_TABLE = """
+    CREATE TABLE IF NOT EXISTS {table} (
+        timestamp TIMESTAMP NOT NULL SORTKEY ENCODE lzo,
+        {schema}
+    );
+""".format(table=TABLE_NAMES["perm"], schema="{schema}")
+
+Q_GET_MAX_DAY = """
+    SELECT MAX(timestamp)::DATE FROM {table};
+""".format(table=TABLE_NAMES["perm"].format(event_type="{event_type}",
+                                            suffix=""))
+
+Q_CHECK_FOR_DAY = """
+    SELECT timestamp FROM {table}
+    WHERE timestamp::DATE = '{day}'::DATE
+    LIMIT 1;
+""".format(table=TABLE_NAMES["perm"].format(event_type="{event_type}",
+                                            suffix="_sampled_10"),
+           day="{day}")
+
+Q_CREATE_CSV_TABLE = """
+    CREATE TABLE IF NOT EXISTS {table} (
+        timestamp BIGINT NOT NULL SORTKEY,
+        {schema}
+    );
+""".format(table=TABLE_NAMES["temp"], schema="{schema}")
+
+Q_COPY_CSV = """
+    COPY {table} (
+        timestamp,
+        {columns}
+    )
+    FROM '{s3_path}'
+    CREDENTIALS 'aws_access_key_id={aws_access_key_id};aws_secret_access_key={aws_secret_access_key}'
+    FORMAT AS CSV
+    TRUNCATECOLUMNS;
+""".format(table=TABLE_NAMES["temp"],
+           columns="{columns}",
+           s3_path="{s3_path}",
+           aws_access_key_id="{aws_access_key_id}",
+           aws_secret_access_key="{aws_secret_access_key}")
+
+Q_CLEAR_DAY = """
+    DELETE FROM {table}
+    WHERE timestamp::DATE = '{day}'::DATE;
+""".format(table=TABLE_NAMES["perm"], day="{day}")
+
+Q_INSERT_EVENTS = """
+    INSERT INTO {perm_table} (timestamp, {columns})
+    SELECT ts, {columns}
+    FROM (
+        SELECT
+            *,
+            'epoch'::TIMESTAMP + timestamp * '1 second'::INTERVAL AS ts,
+            STRTOL(SUBSTRING({id_column} FROM 0 FOR 8), 16) % 100 AS cohort
+        FROM {temp_table}
+    )
+    WHERE cohort <= {percent}
+    AND ts::DATE = '{day}'::DATE
+    AND ts::DATE >= '{max_day}'::DATE - '{months} months'::INTERVAL;
+""".format(perm_table=TABLE_NAMES["perm"],
+           temp_table=TABLE_NAMES["temp"],
+           columns="{columns}",
+           id_column="{id_column}",
+           percent="{percent}",
+           day="{day}",
+           max_day="{max_day}",
+           months="{months}")
+
+Q_GET_TIMESTAMP = """
+    SELECT {which}(timestamp) FROM {table};
+""".format(which="{which}", table=TABLE_NAMES["temp"])
+
+Q_DELETE_EVENTS = """
+    DELETE FROM {table}
+    WHERE timestamp::DATE < '{day}'::DATE - '{months} months'::INTERVAL;
+""".format(table=TABLE_NAMES["perm"], day="{day}", months="{months}")
+
+Q_VACUUM_TABLES = """
+    END;
+    VACUUM FULL {table};
+    ANALYZE {table};
```

Comment:
Fwiw, we weren't doing this `ANALYZE` step previously, so weren't getting all of the optimization benefit available to us.